modules/travel_pay/app/services/travel_pay/claims_service.rb

```diff
@@ -21,6 +21,31 @@ def get_claims(params = {})
       }
     end
 
+    def get_claims_by_date_range(params = {})
+      if params['start_date'] && params['end_date']
+        DateTime.parse(params['start_date'].to_s) && DateTime.parse(params['end_date'].to_s)
```

Comment:
What's this line for - validation?

Reply:
So yes, to make sure both the start time and end time are provided, and that they're valid dates. Had to do them in the same place because the `DateTime.parse` kept failing if the the end time was `nil` so I put that in as a check to make sure it was present (in both places) before trying to parse it to validate it (and the rescue Date::Error below would catch if the `parse` failed due to it not being a valid datetime).

Reply:
It looked a little confusing to my untrained Ruby eyes. I wonder if it would be good to break the validation logic & error raising to a method we could call from here. That way it's clear what this is doing

Reply:
My "untrained Ruby eyes" just made it up - I have no idea what the best approach would be, I just wanted it to stop throwing up all over the tests. 🤣 

But, I can look at extracting that, sure. 😁

Reply:
@kjduensing I separated that "validate" step out into a separate function, but now I'm wondering if we want to move it even further out and have like a shared "validate the date" helper function that checks for valid dates any time we pass in a date for a param.... 🤔

Reply:
Looks like that's a JS package, so we'd need to find a Ruby version, though. 🤔 

You ok to go ahead and merge as is (so we can incorporate this into the appt association logic PR) while we're looking though? (I can make a separate followup ticket for tracking purposes.)

Reply:
🤦

Reply:
yes, let's merge.